controllers/apps/operations/datascript.go

```diff
@@ -219,177 +212,177 @@ func getScriptContent(reqCtx intctrlutil.RequestCtx, cli client.Client, spec *ap
-			ImagePullPolicy: corev1.PullPolicy(viper.GetString(constant.KBImagePullPolicy)),
-			Command:         jobCmdTpl,
-			Env:             envs,
-		}
-		randomStr, _ := password.Generate(4, 0, 0, true, false)
-		jobName := fmt.Sprintf("%s-%s-%s-%s", cluster.Name, "script", ops.Name, randomStr)
-		if len(jobName) > 63 {
-			jobName = strings.TrimSuffix(jobName[:63], "-")
-		}
-
-		job := &batchv1.Job{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      jobName,
-				Namespace: cluster.Namespace,
-			},
-		}
-		intctrlutil.InjectZeroResourcesLimitsIfEmpty(&container)
-		// set backoff limit to 0, so that the job will not be restarted
-		job.Spec.BackoffLimit = pointer.Int32(0)
-		job.Spec.Template.Spec.RestartPolicy = corev1.RestartPolicyNever
-		job.Spec.Template.Spec.Containers = []corev1.Container{container}
-		job.Spec.Template.Spec.ImagePullSecrets = intctrlutil.BuildImagePullSecrets()
-
-		// add labels
-		job.Labels = getDataScriptJobLabels(cluster.Name, component.Name, ops.Name)
-		// add tolerations
-		schedulingPolicy, err := scheduling.BuildSchedulingPolicy(cluster, component)
-		if err != nil {
-			return nil, intctrlutil.NewFatalError(err.Error())
-		}
-		job.Spec.Template.Spec.Tolerations = schedulingPolicy.Tolerations
-		// add owner reference
-		scheme, _ := appsv1alpha1.SchemeBuilder.Build()
-		if err := controllerutil.SetOwnerReference(ops, job, scheme); err != nil {
-			return nil, intctrlutil.NewFatalError(err.Error())
-		}
-		return job, nil
-	}
-
-	// parse kb host
-	var endpoint string
-	var job *batchv1.Job
-
-	jobs := make([]*batchv1.Job, 0)
-	if ops.Spec.ScriptSpec.Selector == nil {
-		if endpoint, err = getTargetService(reqCtx, cli, client.ObjectKeyFromObject(cluster), component.Name); err != nil {
-			return nil, intctrlutil.NewFatalError(err.Error())
-		}
-		if job, err = buildJob(endpoint); err != nil {
-			return nil, intctrlutil.NewFatalError(err.Error())
-		}
-		jobs = append(jobs, job)
-		return jobs, nil
-	}
-
-	selector, err := metav1.LabelSelectorAsSelector(ops.Spec.ScriptSpec.Selector)
-	if err != nil {
-		return nil, intctrlutil.NewFatalError(err.Error())
-	}
-
-	pods := &corev1.PodList{}
-	if err = cli.List(reqCtx.Ctx, pods, client.InNamespace(cluster.Namespace),
-		client.MatchingLabels{
-			constant.AppInstanceLabelKey:    cluster.Name,
-			constant.KBAppComponentLabelKey: component.Name,
-		},
-		client.MatchingLabelsSelector{Selector: selector},
-	); err != nil {
-		return nil, intctrlutil.NewFatalError(err.Error())
-	} else if len(pods.Items) == 0 {
-		return nil, intctrlutil.NewFatalError(err.Error())
-	}
-
-	for _, pod := range pods.Items {
-		endpoint = pod.Status.PodIP
-		if job, err = buildJob(endpoint); err != nil {
-			return nil, intctrlutil.NewFatalError(err.Error())
-		} else {
-			jobs = append(jobs, job)
-		}
-	}
-	return jobs, nil
-}
+// func getTargetService(reqCtx intctrlutil.RequestCtx, cli client.Client, clusterObjectKey client.ObjectKey, componentName string) (string, error) {
```

Comment:
can be deleted

Reply:
https://github.com/apecloud/kubeblocks/pull/7968